tower-http/src/redirect.rs

```diff
@@ -0,0 +1,100 @@
+//! Service that redirects all requests.
+
+use http::{header, HeaderValue, Response, StatusCode, Uri};
+use std::{
+    convert::Infallible,
+    future::Future,
+    marker::PhantomData,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_service::Service;
+
+/// Service that redirects all requests.
+#[derive(Clone, Debug)]
```

Comment:
nit: the derived impls will require `ResBody` to be `Debug` and `Clone`. i think we should probably have manual impls, since this doesn't actually contain a `ResBody`...

Reply:
I'm beginning to think about bringing in some proc-macro crate that has more flexible `Clone`, `Debug, etc, derivation 😅

Reply:
in general, i think "don't derive `Clone` or `Debug` for types with `PhantomData`" is probably a good rule of thumb